django/db/backends/base/validation.py

```diff
@@ -1,10 +1,32 @@
+from django.core import checks
+from django.db import connection
+
+
 class BaseDatabaseValidation:
     """Encapsulate backend-specific validation."""
     def __init__(self, connection):
         self.connection = connection
 
     def check(self, **kwargs):
-        return []
+        return self._check_db_version()
+
+    def _check_db_version(self):
+        errors = []
+        db_version = self.connection.get_version()
+        minimum_database_version = connection.features.minimum_database_version
```

Comment:
I would move the logic of determining if the version is supported to `DatabaseWrapper.is_database_version_supported()`. It will need to be overridden in the case of MySQL where there are different version numbers for MySQL and MariaDB.